### PR TITLE
Fix #3565 profile_progress_recently_played_story_card.xml merged into single xml file

### LINKED ISSUE #3565: Merge profile_progress_recently_played_story_card.xml into single xml file
Currently there are 4 versions of `profile_progress_recently_played_story_card.xml` file merge it into single xml file.

We can use https://text-compare.com/ to compare two versions of this file and for all the differences we can create variables in `dimens.xml` file and use it accordingly.

**Note**: In PR, make sure you add before and after screenshot of mobile-portrait, mobile-landscape, tablet-portrait and tablet-landscape for comparison and make sure that there is not difference between before and after UI.

### PR DESCRIPTION
**Pull request by yashbansal130**
## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes

| Screen Before | Screen After |
| --- | --- |
| before_port | after_port |
| before_land | after_land |
| Screenshot_1644648185 | after-land-tab |
| before-port-tab | after-port-tab |

- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by yash10019coder** [comment]
HI @yashbansal130 thanks for creating the PR I suggest you to add screenshots in tabular form [you can look at this PR for reference](https://github.com/oppia/oppia-android/pull/3523) and @rt4914 please approve the workflow

**Comment by yashbansal130** [comment]
@yash10019coder looks good?

**Review by yash10019coder — CHANGES_REQUESTED** [review]
Hi @yashbansal130 you have done almost all the work I have left a comment thanks

**Comment by yashbansal130** [comment]
@yash10019coder  sorry but no change request is visible to me nor any comment.

File: app/src/main/res/layout/profile_progress_recently_played_story_card.xml
Comment (yash10019coder) [inline app/src/main/res/layout/profile_progress_recently_played_story_card.xml]: why we have to add a scaleType here

**Review by yash10019coder — CHANGES_REQUESTED** [review]
@yashbansal130 why do we have to add dimensions values for layout-land swp600 and land-swp600 as their layout files are deleted so where are they used

File: app/src/main/res/layout/profile_progress_recently_played_story_card.xml
Comment (yashbansal130) [reply app/src/main/res/layout/profile_progress_recently_played_story_card.xml]: because it was added in sw600dp0-land so just for being sure its absence doesnt create a problem

**Comment by yashbansal130** [comment]
> @yashbansal130 why do we have to add dimensions values for layout-land swp600 and land-swp600 as their layout files are deleted so where are they used

they have the same name as other dimen for layout_profile_recently_played_activity so what I know is that it will automatically refer to this dimen depending upon the layout is land/port/sw600dp-land/sw600dp-port

**Comment by rt4914** [comment]
Will review it once @yash10019coder has approved it.

File: app/src/main/res/layout/profile_progress_recently_played_story_card.xml
Comment (yash10019coder) [reply app/src/main/res/layout/profile_progress_recently_played_story_card.xml]: but it was for tablet and now it will also be applied to mobile

**Comment by yash10019coder** [comment]
> > @yashbansal130 why do we have to add dimensions values for layout-land swp600 and land-swp600 as their layout files are deleted so where are they used
> 
> they have the same name as other dimen for layout_profile_recently_played_activity so what I know is that it will automatically refer to this dimen depending upon the layout is land/port/sw600dp-land/sw600dp-port

ok I get that why are we declaring dimes for other configurations

File: app/src/main/res/layout/profile_progress_recently_played_story_card.xml
Comment (rt4914) [reply app/src/main/res/layout/profile_progress_recently_played_story_card.xml]: We don't need `android:scaleType="fitCenter"` in `LessonThumbnailImageView `because it has already been implemented in kotlin files. https://github.com/oppia/oppia-android/blob/3a0afb4eb8222d4ff3acd2e2e660b85911d35229/app/src/main/java/org/oppia/android/app/customview/LessonThumbnailImageView.kt#L110

Comment (yashbansal130) [reply app/src/main/res/layout/profile_progress_recently_played_story_card.xml]: ok i will delete it

**Review by yash10019coder — COMMENTED** [review]
@yashbansal130 please update the screenshots.

**Comment by yash10019coder** [comment]
and also resolve the merge confilcts.

**Comment by yashbansal130** [comment]
@yash10019coder 
to get new SS, i did fetch and merge and git pull to get the updated code because i was 10 commits behind the develop branch, after doing that my emulator is not showing the whole recyclerview of the recently_played_story_card
Screenshot from 2022-02-02 21-55-03

**Comment by yash10019coder** [comment]
This is strange I'll look into this and @rt4914 can you please approve the workflow here thanks

**Comment by yash10019coder** [comment]
and also yash remember to assign the reviewers if you are asking something from them it lets us know that we have to see this PR

**Comment by yashbansal130** [comment]
> and also yash remember to assign the reviewers if you are asking something from them it lets us know that we have to see this PR

i will keep this in mind surely

**Comment by yashbansal130** [comment]
> This is strange I'll look into this and @rt4914 can you please approve the workflow here thanks

any updates?

**Comment by yashbansal130** [comment]
> This is strange I'll look into this and @rt4914 can you please approve the workflow here thanks

@yash10019coder any updates?

**Comment by yash10019coder** [comment]
I think that your branch is not up to date with develop you must merge with latest develop thanks

**Comment by yashbansal130** [comment]
@yash10019coder updated screenshots.

**Review by yash10019coder — APPROVED** [review]
@yashbansal130 LGTM thanks for your work

**Comment by yashbansal130** [comment]
@yash10019coder can i do some changes i think something is missing

**Comment by yashbansal130** [comment]
@yash10019coder actually the dimen files which are declared somehow got reset when i removed the scaleType attribute in the xml layout file, i just noticed that it was not appearing in the commits, made the necessory changes.

**Review by yash10019coder — APPROVED** [review]
@yashbansal130  okay so while reverting scaleType all the dimens all got reverted and as have you added them again so LGTM

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Review by rt4914 — CHANGES_REQUESTED** [review]
Blocking as I might need to check this again.

**Comment by yashbansal130** [comment]
@rt4914 did u requested changes, because I am not able to see the changes you requested? Thanks

**Review by rishidyno — CHANGES_REQUESTED** [review]
@yashbansal130 please add SS properly and use Emulator screenshot feature to take SS.
Also, I can see mobile-port and land SS are not aligned properly.
image  use this button to take SS.

**Comment by yashbansal130** [comment]
I am getting invalid tag error for tablet even after uninstalling the app and building the project again and again installing the app. First the profile page opens when there is no recently played stories but after going through 1-2 stories the profile activity crashes. I tried this 3 times

Screenshot from 2022-02-11 20-37-34

**Comment by yashbansal130** [comment]
updated the SS for mobile

**Comment by yash10019coder** [comment]
try to uninstall the app again then clean the project under build tab and then rebuild the project

**Comment by rishidyno** [comment]
> updated the SS for mobile

Still mobile-port are not correct.

**Comment by yashbansal130** [comment]
> > updated the SS for mobile
> 
> Still mobile-port are not correct.

whats the issue, i actually scrolled that a little

**Comment by rishidyno** [comment]
> > > updated the SS for mobile
> > 
> > 
> > Still mobile-port are not correct.
> 
> whats the issue, i actually scrolled that a little

Yup! Always make sure that they are taken in same setting.

**Comment by yashbansal130** [comment]
> Yup

ok so i dont have to scroll it and align both the ss?

**Comment by rishidyno** [comment]
> > Yup
> 
> ok so i dont have to scroll it and align both the ss?

Yup

**Comment by yash10019coder** [comment]
> > > updated the SS for mobile
> > 
> > 
> > Still mobile-port are not correct.
> 
> whats the issue, i actually scrolled that a little

the issue is like we want to check whether the ss's are same or not 
whatever we are doing should reflect no changes in the ui

**Comment by yashbansal130** [comment]
> the issue is like we want to check whether the ss's are same or not whatever we are doing should reflect no changes in the ui

updated all SS

**Review by yash10019coder — APPROVED** [review]
@yashbansal130 LGTM thanks and @rt4914 can you please approve the workflow

**Comment by rishidyno** [comment]
LGTM Thanks @yashbansal130.

**Review by rishidyno — APPROVED** [review]

**Comment by rt4914** [comment]
Re-running at all CI checks were failing. If they do not pass please merge with latest develop again @yashbansal130

**Comment by yashbansal130** [comment]
@rt4914 resolved conflicts

File: app/src/main/res/values-land/dimens.xml
Comment (rt4914) [inline app/src/main/res/values-land/dimens.xml]: Remove this blank line

File: app/src/main/res/values/dimens.xml
Comment (rt4914) [inline app/src/main/res/values/dimens.xml]: Remove this blank line

File: app/src/main/res/values-land/dimens.xml
Comment (rt4914) [inline app/src/main/res/values-land/dimens.xml]: Bottom margin should be `0dp` for landscape please check this and confirm once again.

File: app/src/main/res/values-sw600dp-land/dimens.xml
Comment (rt4914) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: This should be `32dp`

File: app/src/main/res/values-sw600dp-land/dimens.xml
Comment (rt4914) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: This should be `28dp`

File: app/src/main/res/values-sw600dp-land/dimens.xml
Comment (rt4914) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: This should be `0dp`

File: app/src/main/res/values-sw600dp-port/dimens.xml
Comment (rt4914) [inline app/src/main/res/values-sw600dp-port/dimens.xml]: This should be `0dp`

**Review by rt4914 — CHANGES_REQUESTED** [review]
@yashbansal130 Suggested changes.

@rishidyno @yash10019coder Please have a look at my comments and assign me once they have been finished correctly. Thanks.

**Review by rishidyno — APPROVED** [review]
@yashbansal130 LGTM, thanks.

**Comment by yash10019coder** [comment]
> @yashbansal130 Suggested changes.
> 
> @rishidyno @yash10019coder Please have a look at my comments and assign me once they have been finished correctly. Thanks.

@yashbansal130 please always reply on comments so that we can know that you have resolved them

**Review by yash10019coder — APPROVED** [review]
LGTM @yashbansal130  thanks

**Comment by rishidyno** [comment]
@rt4914 please aproov the workflows.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks

**Pull request merged**
